CIPs/0008.md

```diff
@@ -64,14 +63,56 @@ I considered doing storing all data on IPFS, however the tooling around IPFS as
 
 More reliance on off-chain data and misunderstanding such reliance could lead to inconsistency or poor usability. Ultimately this is traded off against cost of storing data on-chain or in centralized services.
 
-## Useful Links
+## Specification
+
+### Signatures
+
+All file contents should always be signed by an authorized signer of the account to prevent tampering by the file hoster. Signatures should be stored along side the payload under `${dataPath}.signature` and to construct these signatures the signature specification [EIP712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md) is used. Where it doesn’t make sense to include the payload itself in the EIP712 object we instead include the keccak256 hash hex encoded.
+
+Signing over the path as well as the message itself prevents messages from being shuffled around by an attacker; a message is only valid in the intended context. The path should be normalized to the UTF8 encoding of the absolute URL path component, as specified in RFC 3986 § 3.3. Note that path does not include the storage root, query parameters, or fragment.
+
+It is worth noting that this does not protect against the adversarial storage provider withholding or returning outdated data. Additionally, this does not protect against the adversary swapping out files. If the order, including the existence, of the data is critical, then the developer should consider falling back to on-chain data, as consensus on the current view of the data is required. A possible partial workaround is to add expiry to the schema.
+
+### Privacy
+
+We commonly expect users to share data only with a set of authorized peers. A profile picture, under the path /profile/picture might be shared only with the user’s friends. Creating and publishing a number of ciphertexts to a related path (e.g. /profile/picture/{recipient address}), each encrypted with a friend's public key would be one way to achieve this. It would ensure data integrity and confidentiality.
```

Comment:
Just formatting the paths as code strings.
```suggestion
We commonly expect users to share data only with a set of authorized peers. A profile picture, under the path `/profile/picture` might be shared only with the user’s friends. Creating and publishing a number of ciphertexts to a related path (e.g. `/profile/picture/{recipient address}`), each encrypted with a friend's public key would be one way to achieve this. It would ensure data integrity and confidentiality.
```